runtime/src/main/java/org/corfudb/util/serializer/KeyDynamicProtobufSerializer.java

```diff
@@ -0,0 +1,135 @@
+package org.corfudb.util.serializer;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.DynamicMessage;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufInputStream;
+import io.netty.buffer.ByteBufOutputStream;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.CorfuStoreMetadata;
+import org.corfudb.runtime.collections.CorfuDynamicKey;
+import org.corfudb.runtime.collections.OpaqueCorfuDynamicRecord;
+import org.corfudb.runtime.exceptions.SerializerException;
+
+import java.io.IOException;
+
+@Slf4j
+public class KeyDynamicProtobufSerializer extends DynamicProtobufSerializer {
```

Comment:
It will be good to add a comment/javadoc explaining that this class is used to serialize/deserialize the key from a byte buffer and construct an OpaqueRecord.

Reply:
Sure. Will do. Thanks!